src/token/src/state.rs

```diff
@@ -67,6 +67,10 @@ impl State {
         &mut self.ledger
     }
 
+    pub fn notifications(&self) -> &PendingNotifications {
```

Comment:
Rather than having the notifications accessible via functions just make the `notifications` field public.
This will prevent possible unnecessary borrow checker bugs in the future (because borrow of `&self` vs `&self.notifications`)